### PR TITLE
feat: support Svelte 4

### PR DESCRIPTION
**Pull request by benmccann**

### CODE REVIEW

**Review by dominikg — APPROVED** [review]

File: packages/kit/package.json
Comment (dummdidumm) [inline packages/kit/package.json:46]:
```suggestion
		"svelte": "^3.54.0 || ^4.0.0-next.0 || ^4.0.0",
```

Comment (benmccann) [reply packages/kit/package.json:46]: I tested and verified it's not necessary. but it does turn out that we need to add `^4.0.0-next.0` to `svelte-hmr`, so I'm going to mark this PR as draft

**Comment by benmccann** [comment]
Put this on hold pending https://github.com/sveltejs/svelte-hmr/pull/70

File: packages/kit/package.json
Comment (dummdidumm) [reply packages/kit/package.json:46]: Oh TIL, I thought that you need it separately

Comment (benmccann) [reply packages/kit/package.json:46]: Actually, I realized I tested the wrong thing. I still think it's not necessary, but we should confirm before merging this to make sure

Comment (benmccann) [reply packages/kit/package.json:46]: Yeah, we only need the one. Explained on https://www.npmjs.com/package/semver

**Pull request merged**
